subsys/net/ip/l2/ieee802154/ieee802154_utils.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief IEEE 802.15.4 Management
+ */
+
+#ifndef __IEEE802154_UTILS_H__
+#define __IEEE802154_UTILS_H__
+
+#include <net/ieee802154_radio.h>
+
+static inline void ieee802154_filter_ieee_addr(struct net_if *iface,
+					       u8_t *ieee_addr)
+{
+	const struct ieee802154_radio_api *radio = iface->dev->driver_api;
+
+	if (radio->get_capabilities(iface->dev) & IEEE802154_HW_FILTER) {
+		struct ieee802154_filter filter;
+
+		filter.ieee_addr = ieee_addr;
+
+		if (radio->set_filter(iface->dev,
+				      IEEE802154_FILTER_TYPE_IEEE_ADDR,
+				      &filter) != 0) {
+			NET_WARN("Could not apply IEEE address filter");
+		}
+	}
+}
+
+static inline void ieee802154_filter_short_addr(struct net_if *iface,
+						u16_t short_addr)
+{
+	const struct ieee802154_radio_api *radio = iface->dev->driver_api;
+
+	if (radio->get_capabilities(iface->dev) & IEEE802154_HW_FILTER) {
+		struct ieee802154_filter filter;
+
+		filter.short_addr = short_addr;
+
+		if (radio->set_filter(iface->dev,
+				      IEEE802154_FILTER_TYPE_SHORT_ADDR,
+				      &filter) != 0) {
+			NET_WARN("Could not apply short address filter");
+		}
+	}
+}
+
+static inline void ieee802154_filter_pan_id(struct net_if *iface,
```

Comment:
Why is this needed when there was already a radio->set_pan_id()? Are we planning on abstracting all fot he radio functions behind an ieee802154_<func>? Because it looks a bit strange to see code mixing the two when they ultimately touch a radio function.

Reply:
It disappears in further patch. I did this order of patches for proper API deprecation

Reply:
Is the intent then to add ieee802154_<func>() interface routines for all of the function points defined in ieee802154_radio_api structure? I would recommend doing this because it is a bit confusing to have a mix of access paths.

Reply:
No, it's just fine as it is right now. A structure which has already xyz_ as prefix  does not need to repeat that prefix on its attribute.

Reply:
I don't think you understood my question. What I'm wondering is if the long term intent is to have management functions that abstract the interface to all of the function points defined in ieee802154_radio_api structure?

Reply:
Indeed I misunderstood.

Hum about management functions I guess you are not talking about net_mgmt. (There would be no use to expose the radio API elsewhere than in the L2)

Do you mean having function ieee802154_foo(iface, ...params...) { iface->dev->radio_api->foo(...params...); } ?

If so, it really depends on the use cases. I did that for filtering because it was easier to have those instead of repeating most of the code here and there. It really depends.

Reply:
Yes. That is what I was initially referring to because I find it a bit confusing to see two different methods for interacting with the radio; these ieee802154_foo() functions and the radio_api->foo functions. There was at least one place in the code where the two methods are next to each other.

I think that standardizing on one method is good as it will help future folks grep the code logic. It is also what I thought you were setting out to do with this initial set of changes.

Reply:
> There was at least one place in the code where the two methods are next to each other.

Accessing the same radio function?

This patch was only about filters. But why not generalizing for the others.
It's nicer to read ieee802154_set_channel() than radio->set_channel() yes. Will see in another patch.